### PR TITLE
test(connections): add functional tests for CSFLE

### PR DESCRIPTION
**Pull request by mcasimir**

### CODE REVIEW

File: packages/compass-components/src/components/editor.tsx [hunk 1]
```diff
@@ -51,6 +51,7 @@ type EditorProps = {
   options?: Omit<IAceOptions, 'readOnly'>;
   readOnly?: boolean;
   completer?: unknown;
+  dataTestId?: string;
```
Comment (gribnoysup) [inline packages/compass-components/src/components/editor.tsx:54]: Do you mind changing this to `data-testid` similar to how we do it for some other components? I know that it's not consistent everywhere, but aligning on the prop name that would actually be used in the tests makes it much easier to discover

File: packages/compass-components/src/components/editor.tsx [hunk 2]
```diff
@@ -73,31 +75,51 @@ function Editor({
...
-    />
+        theme="mongodb"
+        width="100%"
+        value={text}
+        onChange={onChangeText}
+        editorProps={{ $blockScrolling: Infinity }}
+        setOptions={setOptions}
+        readOnly={readOnly}
+        // name should be unique since it gets translated to an id
+        name={aceProps.name ?? `ace-editor-${Date.now()}`}
```
Comment (gribnoysup) [inline packages/compass-components/src/components/editor.tsx:95]: There is `useId` hook that we use for that in other places

File: packages/compass-components/src/components/editor.tsx (hunk 1 above)
Comment (mcasimir) [reply packages/compass-components/src/components/editor.tsx:54]: ah yeah no that would be actually awesome

File: packages/compass-components/src/components/editor.tsx [hunk 3]
```diff
@@ -72,32 +75,54 @@ function Editor({
...
+ * setEditorValue(screen.getByTestId('editor-test-id'), 'my text');
+ * ```
+ */
+function setEditorValue(element: HTMLElement, value: string): void {
+  const container = element.querySelector('.ace_editor');
+  if (!container) {
+    throw new Error('Cannot find editor container');
+  }
+  (window as any).ace.edit(container.id).setValue(value);
+}
```
Comment (gribnoysup) [inline packages/compass-components/src/components/editor.tsx:125]: Neat! I wonder if we can use this in e2e tests for mac as a workaround for broken clipboard in evergreen 🤔

Comment (lerouxb) [reply packages/compass-components/src/components/editor.tsx:125]: Ooh yes this should be helpful.

File: packages/compass-connections/src/index.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+  it('connects to localhost by default', async function () {
+    await expectToConnectWith({
+      connectionString: 'mongodb://localhost:27017/?appName=Test+App',
+    });
+  });
+
+  describe('In-Use Encryption', function () {
+    beforeEach(async function () {
+      if (process?.env?.COMPASS_CSFLE_SUPPORT !== 'true') {
+        this.skip();
```
Comment (addaleax) [inline packages/compass-connections/src/index.spec.tsx:109]:
```suggestion
        return this.skip();
```

File: packages/compass-connections/src/index.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+      await expectToConnectWith({
+        connectionString: 'mongodb://localhost:27017/?appName=Test+App',
+        fleOptions: {
+          storeCredentials: false,
+          autoEncryption: {
+            keyVaultNamespace: 'db.coll',
+            encryptedFieldsMap: {
+              '$compass.error': null,
+              '$compass.rawText': '{ "db.coll": { fields: [] } }',
+              'db.coll': { fields: [] },
```
Comment (addaleax) [inline packages/compass-connections/src/index.spec.tsx:166]: I think one of the things that we should check here if we can is that this actually allows including BSON values, not just JSON, e.g. `fields: [{path: 'foo', bsonType: 'string', keyId: UUID('11d58b8a-0c6c-4d69-a0bd-70c6d9befae9') }]`

**Review by addaleax — APPROVED** [review]
this is awesome! apart from ideally moving the tests, only one comment :)

File: packages/compass-connections/src/index.spec.tsx (hunk 5 above)
Comment (mcasimir) [reply packages/compass-connections/src/index.spec.tsx]: Tested the `UUID`, is there any known reason why `expect(keyid).to.deep.equal(new bson.UUID('11d58b8a-0c6c-4d69-a0bd-70c6d9befae9'))` didn't work while `Binary` did instead?

Comment (addaleax) [reply packages/compass-connections/src/index.spec.tsx]: Yeah, `bson.UUID` is actually not a BSON type, it’s more like a generic UUID class. `new bson.UUID(...).toBinary()` should work :slightly_smiling_face:

**Review by addaleax — APPROVED** [review]

**Pull request merged**